### DISCUSSION TITLE
Multiple Presets + Nuget registries

### DISCUSSION THREAD

**Discussion by adam-clauss** [renovatebot/renovate#29171] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket Server / Renovate 37.368.9

### Please tell us more about your question or problem

We are using presets to configure alternative nuget registry URLs that some of our projects need for private repos.  We're not seeing expected behavior when it comes to these getting combined.  Specifically, PresetA adds:
"nuget": {
  "registryUrls": [
     "https://path-to-nuget-A"
}

Then a very similar for PresetB:
"nuget": {
  "registryUrls": [
     "https://path-to-nuget-B"
}

However, when a specific repo is setup with a configuration that extends both of these, it doesn't look like both are taking effect.  It might only be the last extends.  We are seeing warnings in the renovate logs "Failed to look up nuget package ..." for packages that are located in the private registry that was extended first.

Is there a way to get the registryUrls to become additive rather than replacing each other?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Registry URLs aren't mergeable. If they were then nobody would be able to override them.

You can use packageRules for more granular matching and set registryUrls that way too

**Reply by adam-clauss** [reply]
Unfortunately, due to some of the registries being licensed, we do not want to expose this URL to the individual projects.  Each user has a user-specific URL they use + one used by the CI process.  We don't want the CI URL visible to projects - we did that once before and it caused licensing problems when some people started using it accidentally.
